### ISSUE TITLE
Graves replace lava then you lose all your items

### ISSUE THREAD

**Issue by Peach774** [FireBall1725/Graves#50]
Because a grave replaces blocks, when you die in lava, it replaces the lava and when you break the grave you get your hotbar back and lose the rest of your items because they burn up in lava. It would be nice if when you break a grave that replaced lava, the lava did not come back so you could get all your items, otherwise a gravestone is essentially useless.

**Comment by IrationalReticulatingAugment** [comment]
This just happened to me as well, was rather unexpected.

**Comment by FusionLord** [comment]
What version? Why didn't you fill in the lava?

**Comment by IrationalReticulatingAugment** [comment]
I can get you version info when I get home (at work at the moment).

When I've broken graves in the past, I never noticed items actually ending
up in the world. Usually when I break something holding items (e.g. vanilla
chests), then end up strewn all over. I never noticed this behavior with
the graves; it seemed like the items were just being restored to my
inventory without ever entering the world. The exception to this was
overflow, which ended up in a neat pile at my feet the one time I didn't
have space for everything.

When I took a deep swim in lava on accident and managed to swim to just
below the surface before dying, my grave ended up 1 block below the surface
of the lava. Through clever use of cobblestone, I managed to find it and
broke it, expecting the items to be placed in my inventory. I got my
hotbar, but not the rest of my inventory.

I was very confused, to the say the least. Now I know how it works, I'll be
more careful retrieving items in the future, but it sucks I had to find out
this way.

**Comment by FusionLord** [comment]
When you break the grave the items go directly into your inventory aside from the overflow which goto your feet. So if they got thrown all around I need the version number to find out why.

**Comment by IrationalReticulatingAugment** [comment]
To be clear, I've never had a grave spew items everywhere, I've only had it
place items in my inventory with the overflow at my feet, like you say
should happen. I suppose the bug is that breaking a grave surrounded/in
lava causes loss of inventory/armor slots somehow, but not hotbars.

I'll be home from work in about 5 hours and I'll get you version info then
(Downloaded in the last couple days, likely b21 or 22, but I'll confirm.
I'm on 1.10), and I'll try to replicate the problem.

**Comment by IrationalReticulatingAugment** [comment]
graves-1.10.2-b21-universal
With forge: release 1.10.2-forge1.10.2-12.18.1.2014.

I tried replicating the problem on a survival world with cheats enabled. 

a 2 deep laval pool, grave at the bottom, broke and restored all items. 
a 3 deep pool of lava, grave at the bottom, restored all items.
a 3 deep pool of lava, grave in the middle (so lava above, on all sides, and below): restored all items. 
a 3 deep pool of lava, grave in the middle, but with lava above removed: restored all items.

**Comment by FusionLord** [comment]
So what you are saying is that from all you tests is that it is working as intended?

**Comment by IrationalReticulatingAugment** [comment]
Yes. And I just had it happen to me again. I died, I came back, I died to
zombie, I came back again, and a bunch of items were missing from the grave
when I broke it, and there was no grave from when the zombie killed me. No
lava involved this time.

I'm trying to replicate it, but haven't had any luck so far. Going to
update and hope for the best. If it happens again I'll keep better notes.

**Comment by FusionLord** [comment]
ok That is a different issue. are these the same steps take the first time... items that are in your inventory when you die to the zombie go into the existing grave... so the first time did you die to the zombie?

**Comment by FusionLord** [comment]
@sinergistic could you post a log after reproducing those steps.

**Comment by IrationalReticulatingAugment** [comment]
Sorry for the slow reply.

Thinking back, I believe I did die to the zombie both times this happened,
but didn't think much of it the first time. I wasn't able to reproduce the
bug in a separate world successfully, but I can try again tonight. In the
meantime I'll be updating to the latest version in the hopes I don't lose
anything else.

My steps for trying to reproduce went as follows, if you're curious:

Edit graves config to 100% guarantee a zombie spawn.
Survival amplified world with cheats enabled.
Move out of chunk loading range of spawn using creative /tp (both times I
died, I died in chunks that should have unloaded.)
Use creative mode to fill my inventory up with random items.
Fall to my death
Return to grave, break grave in survival mode
Die to zombie
Return to grave, break grave in survival mode
Check items. I tested it several times and always ended up with all the
items in my inventory.

I probably wasn't as thorough as I should have been. I don't think I had
armor on any of the times, and I was mostly just using random blocks to
fill up my inventory, instead of random mod items.

If it happens again, what logs do you want? Just the stuff from the console?

**Comment by FusionLord** [comment]
I would like the Latest log file, it should have the word "latest" in the file name.
